rdmo/projects/permissions.py

```diff
@@ -8,11 +8,21 @@ def has_permission(self, request, view):
         if not (request.user and request.user.is_authenticated):
             return False
 
-        # always return True:
-        # for retrieve, update, partial_update, the permission will be checked on the
-        # object level (in the next step), list and create is allowed for every user since
-        # the filtering is done in the queryset
-        return True
+        if view.detail:
```

Comment:
this doesn't need to be in a tree when you use `return` after every `if`, please keep it flat for readability.
```py
if view.detail:
    # for retrieve, update, partial_update, the permission will be checked on the
    # object level (in the next step)
    return True

if view.action == 'list':
    # list is allowed for every user since the filtering is done in the queryset
    return True

if 'create' in view.action_map.values():
    # for create, check the permission (from rules.py),
    # but only if it is not a ReadOnlyValueSet (i.e. only for ProjectViewSet)
    return super().has_permission(request, view)

return True
```
or not?

Reply:
🙏